Actually, I just noticed that there are some type errors which caused the CI to fail for this PR. I don't think the CI server is visible to non-CR employees, so pasted below:

```
ccl/src/views/clusterviz/containers/map/mapLayout.tsx(204,10): error TS2322: Type '{ ref: RefObject<Node>; children: (Element | Element[])[]; }' is not assignable to type 'SVGProps<SVGGElement>'.
[17:12:30]  Types of property 'ref' are incompatible.
[17:12:30]    Type 'RefObject<Node>' is not assignable to type 'Ref<SVGGElement>'.
[17:12:30]      Type 'RefObject<Node>' is not assignable to type 'RefObject<SVGGElement>'.
[17:12:30]        Type 'Node' is not assignable to type 'SVGGElement'.
[17:12:30]          Property 'farthestViewportElement' is missing in type 'Node'.
[17:12:30]ccl/src/views/clusterviz/containers/map/nodeCanvas.tsx(150,13): error TS2322: Type '{ style: { width: string; height: string; marginBottom: number; position: "absolute"; }; classNam...' is not assignable to type 'SVGProps<SVGSVGElement>'.
[17:12:30]  Types of property 'ref' are incompatible.
[17:12:30]    Type 'RefObject<SVGElement>' is not assignable to type 'Ref<SVGSVGElement>'.
[17:12:30]      Type 'RefObject<SVGElement>' is not assignable to type 'RefObject<SVGSVGElement>'.
[17:12:30]        Type 'SVGElement' is not assignable to type 'SVGSVGElement'.
[17:12:30]          Property 'contentScriptType' is missing in type 'SVGElement'.
[17:12:30]src/views/cluster/components/linegraph/index.tsx(197,16): error TS2322: Type '{ className: string; color?: string; height?: string | number; id?: string; lang?: string; max?: ...' is not assignable to type 'SVGProps<SVGSVGElement>'.
[17:12:30]  Types of property 'ref' are incompatible.
[17:12:30]    Type 'string | RefObject<SVGElement> | ((instance: SVGSVGElement) => any) | RefObject<SVGSVGElement>' is not assignable to type 'Ref<SVGSVGElement>'.
[17:12:30]      Type 'RefObject<SVGElement>' is not assignable to type 'Ref<SVGSVGElement>'.
[17:12:30]        Type 'RefObject<SVGElement>' is not assignable to type 'RefObject<SVGSVGElement>'.
[17:12:30]src/views/shared/components/sql/box.tsx(27,12): error TS2322: Type '{ className: "sql-highlight"; ref: RefObject<Node>; children: string; }' is not assignable to type 'DetailedHTMLProps<HTMLAttributes<HTMLPreElement>, HTMLPreElement>'.
[17:12:30]  Type '{ className: "sql-highlight"; ref: RefObject<Node>; children: string; }' is not assignable to type 'ClassAttributes<HTMLPreElement>'.
[17:12:30]    Types of property 'ref' are incompatible.
[17:12:30]      Type 'RefObject<Node>' is not assignable to type 'Ref<HTMLPreElement>'.
[17:12:30]        Type 'RefObject<Node>' is not assignable to type 'RefObject<HTMLPreElement>'.
[17:12:30]          Type 'Node' is not assignable to type 'HTMLPreElement'.
[17:12:30]            Property 'width' is missing in type 'Node'.
```

I'm not sure what the root issues are here; think you could investigate? You can run the typechecker locally with `make ui-lint` to verify that they are fixed.

--- a/pkg/ui/ccl/src/views/clusterviz/containers/map/mapLayout.tsx
+++ b/pkg/ui/ccl/src/views/clusterviz/containers/map/mapLayout.tsx
@@ -37,7 +37,7 @@ interface MapLayoutState {
 }
 
 export class MapLayout extends React.Component<MapLayoutProps, MapLayoutState> {
-  gEl: any;
+  gEl: React.RefObject<SVGGElement> = React.createRef();
   zoom: d3.behavior.Zoom<any>;
   maxLatitude = 80;
 
@@ -88,7 +88,7 @@ export class MapLayout extends React.Component<MapLayoutProps, MapLayoutState> {
       // target zoom state. This is needed because free pan-and-zoom does not
       // update the internal animation state used by zoom.event, and will cause
       // animations after the first to have the wrong starting position.
-      d3.select(this.gEl)
+      d3.select(this.gEl.current)
         .call(this.zoom.event)
         .transition()
         .duration(750)
@@ -99,7 +99,7 @@ export class MapLayout extends React.Component<MapLayoutProps, MapLayoutState> {
         );
     } else {
       // Call zoom.event on the element itself, rather than a transition.
-      d3.select(this.gEl)
+      d3.select(this.gEl.current)
         .call(this.zoom
           .scale(zt.scale())
           .translate(zt.translate())
@@ -161,7 +161,7 @@ export class MapLayout extends React.Component<MapLayoutProps, MapLayoutState> {
   }
 
   componentDidMount() {
-    d3.select(this.gEl).call(this.zoom);
+    d3.select(this.gEl.current).call(this.zoom);
     this.rezoomToLocalities(this.state.zoomTransform);
   }
 
@@ -201,7 +201,7 @@ export class MapLayout extends React.Component<MapLayoutProps, MapLayoutState> {
     const { viewportSize } = this.props;
 
     return (
-      <g ref={el => this.gEl = el}>
+      <g ref={this.gEl}>
         <rect width={viewportSize[0]} height={viewportSize[1]} fill="#E2E5EE" />
         <WorldMap projection={projection} />
         { this.renderChildLocalities(projection) }

--- a/pkg/ui/ccl/src/views/clusterviz/containers/map/nodeCanvas.tsx
+++ b/pkg/ui/ccl/src/views/clusterviz/containers/map/nodeCanvas.tsx
@@ -41,7 +41,7 @@ interface NodeCanvasState {
 }
 
 export class NodeCanvas extends React.Component<NodeCanvasProps, NodeCanvasState> {
-  graphEl: SVGElement;
+  graphEl: React.RefObject<SVGSVGElement> = React.createRef();
   debouncedOnResize: () => void;
 
   constructor(props: any) {
@@ -52,7 +52,7 @@ export class NodeCanvas extends React.Component<NodeCanvasProps, NodeCanvasState
   }
 
   updateViewport = () => {
-    const rect = this.graphEl.getBoundingClientRect();
+    const rect = this.graphEl.current.getBoundingClientRect();
     this.setState({
       viewportSize: [rect.width, rect.height],
     });
@@ -154,7 +154,7 @@ export class NodeCanvas extends React.Component<NodeCanvasProps, NodeCanvasState
               position: "absolute",
             }}
             className="cluster-viz"
-            ref={svg => this.graphEl = svg}
+            ref={this.graphEl}
           >
             { this.renderContent(showMap) }
           </svg>

--- a/pkg/ui/src/views/cluster/components/linegraph/index.tsx
+++ b/pkg/ui/src/views/cluster/components/linegraph/index.tsx
@@ -33,8 +33,8 @@ interface LineGraphProps extends MetricsDataComponentProps {
  * axis.
  */
 export class LineGraph extends React.Component<LineGraphProps, {}> {
-  // The SVG Element in the DOM used to render the graph.
-  graphEl: SVGElement;
+  // The SVG Element reference in the DOM used to render the graph.
+  graphEl: React.RefObject<SVGSVGElement> = React.createRef();
 
   // A configured NVD3 chart used to render the chart.
   chart: nvd3.LineChart;
@@ -83,7 +83,7 @@ export class LineGraph extends React.Component<LineGraphProps, {}> {
 
     // To get the x-coordinate within the chart we subtract the left side of the SVG
     // element and the left side margin.
-    const x = e.clientX - this.graphEl.getBoundingClientRect().left - CHART_MARGINS.left;
+    const x = e.clientX - this.graphEl.current.getBoundingClientRect().left - CHART_MARGINS.left;
     // Find the time value of the coordinate by asking the scale to invert the value.
     const t = Math.floor(timeScale.invert(x));
 
@@ -158,7 +158,7 @@ export class LineGraph extends React.Component<LineGraphProps, {}> {
       }
 
       ConfigureLineChart(
-        this.chart, this.graphEl, metrics, axis, this.props.data, this.props.timeInfo, hoverTime,
+        this.chart, this.graphEl.current, metrics, axis, this.props.data, this.props.timeInfo, hoverTime,
       );
     }
   }
@@ -194,7 +194,7 @@ export class LineGraph extends React.Component<LineGraphProps, {}> {
     return (
       <Visualization title={title} subtitle={subtitle} tooltip={tooltip} loading={!data} >
         <div className="linegraph">
-          <svg className="graph linked-guideline" ref={(svg) => this.graphEl = svg} {...hoverProps} />
+          <svg className="graph linked-guideline" ref={this.graphEl} {...hoverProps} />
         </div>
       </Visualization>
     );

--- a/pkg/ui/src/views/shared/components/sql/box.tsx
+++ b/pkg/ui/src/views/shared/components/sql/box.tsx
@@ -8,23 +8,23 @@ interface SqlBoxProps {
 }
 
 export class SqlBox extends React.Component<SqlBoxProps> {
-  preNode: Node;
+  preNode: React.RefObject<HTMLPreElement> = React.createRef();
 
   shouldComponentUpdate(newProps: SqlBoxProps) {
     return newProps.value !== this.props.value;
   }
 
   componentDidMount() {
-    hljs.highlightBlock(this.preNode);
+    hljs.highlightBlock(this.preNode.current);
   }
 
   componentDidUpdate() {
-    hljs.highlightBlock(this.preNode);
+    hljs.highlightBlock(this.preNode.current);
   }
 
   render() {
     return (
-      <pre className="sql-highlight" ref={ (node) => this.preNode = node }>
+      <pre className="sql-highlight" ref={this.preNode}>
         { this.props.value }
       </pre>
     );

--- a/pkg/ui/src/views/shared/util/d3-react.tsx
+++ b/pkg/ui/src/views/shared/util/d3-react.tsx
@@ -9,16 +9,16 @@ type Chart<T> = (sel: d3.Selection<T>) => void;
  */
 export default function createChartComponent<T>(containerTy: string, chart: Chart<T>) {
   return class WrappedChart extends React.Component<T> {
-    containerEl: Element;
+    containerEl: React.RefObject<Element> = React.createRef();
 
     componentDidMount() {
-      d3.select(this.containerEl)
+      d3.select(this.containerEl.current)
         .datum(this.props)
         .call(chart);
     }
 
     shouldComponentUpdate(props: T) {
-      d3.select(this.containerEl)
+      d3.select(this.containerEl.current)
         .datum(props)
         .call(chart);
 
@@ -28,7 +28,7 @@ export default function createChartComponent<T>(containerTy: string, chart: Char
     render() {
       return React.createElement(
         containerTy,
-        { ref: (el) => this.containerEl = el },
+        { ref: this.containerEl },
       );
     }
   };

